src/lambda/gc02_check_password_protection_mechanisms/app.py

```diff
@@ -0,0 +1,258 @@
+    )
+    return eval_cc
+
+
+def is_guard_duty_enabled():
+    try:
+        response = AWS_GUARD_DUTY_CLIENT.list_detectors()
+        detectorIds = response.get("DetectorIds", [])
+        return len(detectorIds) > 0
+    except botocore.exceptions.ClientError as ex:
+        if "BadRequest" in ex.response['Error']['Code']:
+            ex.response['Error']['Message'] = "Failed to fetch detector ids for GuardDuty. Bad Request."
+        else:
+            ex.response["Error"]["Message"] = "InternalError"
+            ex.response["Error"]["Code"] = "InternalError"
+            
+        raise ex
+    
+def list_cloudtrails():
+    """Fetches a list of all trails in the account"""
+    try:
+        response = AWS_CLOUDTRAIL_CLIENT.list_trails()
+        trails = response.get("Trails")
+        next_token = response.get("NextToken")
+        while next_token != None:
+            response = AWS_CLOUDTRAIL_CLIENT.list_trails()
+            trails = trails + response.get("Trails")
+            next_token = response.get("NextToken")
+        return trails
+    except botocore.exceptions.ClientError as ex:
+        if "UnsupportedOperationException" in ex.response['Error']['Code']:
+            ex.response["Error"]["Message"] = "list_trails operation not supported by CloudTrails."
+        elif "OperationNotPermittedException" in ex.response['Error']['Code']:
+            ex.response["Error"]["Message"] = "list_trails operation not permitted."
+        else:
+            ex.response["Error"]["Message"] = "InternalError"
+            ex.response["Error"]["Code"] = "InternalError"
+            
+        raise ex
+
+def trails_are_logging(trails):
+    """Checks the status of the fetched trails"""    
+    for t in trails:
+        try:
+            trail_arn = t.get("TrailARN")
+            status = AWS_CLOUDTRAIL_CLIENT.get_trail_status(Name=trail_arn)
+            if status.get("IsLogging", False):
+                return False
+        except botocore.exceptions.ClientError as ex:
+            logger.error("Error while trying to fetch cloudtrail status.")
+            logger.error(ex)
+            raise ex
+    return True    
+    
+def trails_configured_for_iam_events(trails):
+    """Checks for a trails with the configuration required to capture IAM events and returns a filtered list"""    
+    filtered_trails = []
+    for t in trails:
+        try:
+            trail_arn = t.get("TrailARN")
+            configuration = AWS_CLOUDTRAIL_CLIENT.get_trail(Name=trail_arn)
+            if configuration.get("IncludeGlobalServiceEvents", False) and configuration.get("IsMultiRegionTrail", False) and configuration.get("IsOrganizationTrail", False):
+                filtered_trails.append(t)
+        except botocore.exceptions.ClientError as ex:
+            logger.error("Error while trying to fetch cloudtrail configuration.")
+            logger.error(ex)
+            raise ex
+    return True  
```

Comment:
Looks like this should be returning the list of `filtered_trails` instead of `True`.